nifpga/session.py

```diff
@@ -390,8 +410,272 @@ def read(self):
+        1. Trying to write data that does not conform to boundaries of the
+        defined register, will be coerced and the user will be warned. This is
+        either data outside the maximum and minimum values, or data that is not
+        a multiple of the delta.
+        2. Using Read() and Write() will only return the numerically values of
+        a fixed point register. In order to access the overflow status bit,
+        users must call the overflow attribute on the Register. The register
+        will only have a set overflow status given, the Register has been
+        enabled, and has at least been read from or written to before.
+    """
+    def __init__(self,
+                 session,
+                 nifpga,
+                 bitfile_register,
+                 base_address_on_device):
+        super(_FxpRegister, self).__init__(session,
+                                           nifpga,
+                                           bitfile_register,
+                                           base_address_on_device)
+
+        self._word_length = bitfile_register.word_length
+        self._integer_word_length = bitfile_register.integer_word_length
+        self._radix_point = self._calculate_radix_point()
+        self._overflow_enabled = bitfile_register.overflow
+        self._signed = bitfile_register.signed
+        self._delta = self._calculate_delta(bitfile_register.word_length,
+                                            bitfile_register.integer_word_length)
+
+    @property
+    def overflow(self):
+        """ This property should only set be whenever this register has overflow
+        enabled"""
+        return self._overflow
+
+    @overflow.setter
+    def overflow(self, value):
+        if self._overflow_enabled:
+            self._overflow = value
+
+    def _calculate_delta(self, word_length, integer_word_length):
+        return 2**(integer_word_length - word_length)
+
+    def read(self):
+        """ Reads the entire FXP number as an array U32 bits then combining
+        the array bits into a single datatype, then convert it into python
+        Decimal. The largest expecting transfer would be a 64 bit fixed point
+        with the overflow status enabled (making an array of length 3).
+
+        Returns:
+            data (DataType.value): The data in the register in a python list.
+        """
+        data = super(_FxpRegister, self).read()
+        data = self._combine_array_of_u32_into_one_value(data)
+        return self._convert_from_read_value_to_decimal(data)
+
+    def _combine_array_of_u32_into_one_value(self, data):
+        combinedData = 0
+        for index in range(0, len(self)):
+            combinedData = (combinedData << 32 * index) + data[index]
+        if len(self) > 1:
+            overflow_bit = 1 if self._overflow_enabled else 0
+            total_num_bits = self._word_length + overflow_bit
+            combinedData = combinedData >> (32 * len(self) - total_num_bits)
```

Comment:
I think this area deserves some explanation. I guess it's compensating for earlier code that always shifted by multiples of 32 when the Fxp size is not N*32. Maybe the shift amount should be stored in an intermediate variable (with a good name) to help readability.